### PR TITLE
[JENKINS-33321] Upgrade to new parent pom

### PR DESCRIPTION
**Pull request by armfergom**
Upgrade to new parent pom. Includes many improvements. More info [here](https://github.com/jenkinsci/plugin-pom)

- [x] Upgrade to new parent pom (version 2.3) keeping the same baseline version (1.625).
- [x] Fix Injected test due to upgrade.
- [x] Fix Findbugs issues.
- [x] Fix Javadoc so that maven javadoc plugin doesn't complain.

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -19,6 +19,11 @@
             <comments>All source code is under the MIT license.</comments>
         </license>
     </licenses>
+    
+    <properties>
+      <jenkins.version>1.625</jenkins.version>
+      <java.level>6</java.level>
+    </properties>
```
Comment (recena) [inline pom.xml]: Jenkins 1.625 and JDK 6? Weird. https://jenkins-ci.org/blog/2015/04/06/good-bye-java6/

Comment (andresrc) [inline pom.xml]: Use the profiles at the end of the parent POM as an orientation of the java levels to use (they are LTS boundaries, not exact version in which it change, but serve as an quick guide).

Comment (armfergom) [inline pom.xml]: I checked the version to use in [org.jenkins-ci.jenkins](https://github.com/jenkinsci/pom/blob/4a51f9fa6f129a46bec55817bd09bcba14d42ffd/pom.xml#L585)
But I didn't notice this was overriden by [org.jenkins-ci.main.pom](https://github.com/jenkinsci/jenkins/blob/stable-1.625/pom.xml#L624)

My bad. Updated now.

**Comment by andresrc** [comment]
:bee:

**Comment by recena** [comment]
:bee:

**Comment by armfergom** [comment]
@reviewbybees done

**Comment by armfergom** [comment]
In order to make the build of the plugin stable using baseline 2.0-rc-1, the parent pom to use must be 2.4+ where version 2.6 of the Jenkins Test Harness is picked up. 

Added one commit bumping parent pom from 2.3 to 2.6. 

@reviewbybees esp. @recena @andresrc who initially reviewed the PR

**Comment by andresrc** [comment]
:bee:

**Comment by recena** [comment]
:bee:

**Comment by armfergom** [comment]
@reviewbybees done

**Comment by armfergom** [comment]
Hi @gboissinot, is this PR something that may be considered to be merged? :) Thanks!

**Comment by recena** [comment]
@armfergom I recommend you to sent an email to jenkins-dev mailing list.

**Pull request merged**